Labels added to dayOffset preference dialog seekbar

Arnold2381 opened the pull request:
## Purpose / Description
Mentioned first https://github.com/ankidroid/Anki-Android/issues/8417#issuecomment-816953643

## Fixes
Fixes #8417 

## Approach
A new view was added for the labels.

Added two string resources in the file
![image](https://user-images.githubusercontent.com/46752548/115747231-a9033d00-a3b2-11eb-8dfd-6d08725322c9.png)


## How Has This Been Tested?
Physical device

![image](https://user-images.githubusercontent.com/46752548/115663984-e50cb280-a35e-11eb-940c-1411074179e3.png)


## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

prateek-singh-3212 reviewed (commented):
Everything looks good to me.

david-allison-1 commented on AnkiDroid/src/main/java/com/ichi2/ui/SeekBarPreference.java:
```diff
@@ -141,4 +157,21 @@ protected void onPrepareDialogBuilder(AlertDialog.Builder builder) {
     }
+
+    private void addLabelsBelowSeekBar() {
+        String labels[]={mXLabel,mYLabel};
```
```suggestion
        String labels[] = {mXLabel,mYLabel};
```

david-allison-1 commented on AnkiDroid/src/main/java/com/ichi2/ui/SeekBarPreference.java:
```diff
@@ -141,4 +157,21 @@ protected void onPrepareDialogBuilder(AlertDialog.Builder builder) {
+    private void addLabelsBelowSeekBar() {
+        String labels[]={mXLabel,mYLabel};
+        for (int count = 0; count < 2; count++) {
+            TextView textView = new TextView(mContext);
```
We have a custom TextView class which fixes crashes in the default Android implementation

```
            TextView textView = new FixedTextView(mContext);
```

david-allison-1 commented on AnkiDroid/src/main/res/xml/preferences_reviewing.xml:
```diff
@@ -37,7 +37,9 @@
             app:interval="1"
-            app:min="0" />
+            app:min="0"
+            app:xlabel="Midnight"
```
Extract these to string resources to allow translation

https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#adding-translations

david-allison-1 reviewed (commented):
Looks fantastic! Nothing wrong with the code structure, a few nitpicks and good to go

Arnold2381 commented:
@david-allison-1 I have also added the screenshot of the file name in which I have added the string value as mentioned in the dev guide in the PR template.

david-allison-1 commented on AnkiDroid/src/main/java/com/ichi2/ui/SeekBarPreference.java:
```diff
@@ -141,4 +158,22 @@ protected void onPrepareDialogBuilder(AlertDialog.Builder builder) {
     }
+
+    private void addLabelsBelowSeekBar() {
+        String labels[] = {mContext.getString(R.string.xlabel_dayoffset_seekbar),
```
You should be using `mXLabel` and `mYLabel` here

Arnold2381 replied:
When I used that and set the text as `mXlabel` and `mYlabel` in text view it's giving as id's @2134571.

david-allison-1 replied:
```java
        String labels[] = { mContext.getString(mXLabel), mContext.getString(mYLabel)};
```

david-allison-1 commented on AnkiDroid/src/main/java/com/ichi2/ui/SeekBarPreference.java:
```diff
@@ -141,4 +158,22 @@ protected void onPrepareDialogBuilder(AlertDialog.Builder builder) {
+        for (int count = 0; count < 2; count++) {
+            TextView textView = new FixedTextView(mContext);
+            textView.setText(labels[count]);
+            textView.setGravity(Gravity.LEFT);
```
```suggestion
            textView.setGravity(Gravity.START);
```

Arnold2381 replied:
I think it should be better if kept as `LEFT` because then I have to change the params things too. is there a specific reason you mentioned `START`?

david-allison-1 replied:
What happens with a RTL locale? Presumably the seekbar is reversed, we also want to reverse the labels if this is the case

Arnold2381 replied:
Yea got it! Ok, I will make it work!!

Arnold2381 replied:
@david-allison-1 is this okay? I enabled RTL and checked
![image](https://user-images.githubusercontent.com/46752548/115775553-a9123580-a3d0-11eb-8d51-0370e0eb35b3.png)

david-allison-1 replied:
Looks great! 23 -> 11 and 0 -> midnight

Arnold2381 replied:
Yess!

david-allison-1 commented on AnkiDroid/src/main/java/com/ichi2/ui/SeekBarPreference.java:
```diff
@@ -33,7 +35,8 @@
     private final int mInterval;
     private int mValue = 0;
-
+    private final String mXLabel;
+    private final String mYLabel;
```
```suggestion
    @StringRes
    private final int mXLabel;
    @StringRes
    private final int mYLabel;
```

Arnold2381 replied:
Thanks for this!! I was finding this only. But couldn't. I tried converting it into `int` was crashing.

david-allison-1 commented on AnkiDroid/src/main/java/com/ichi2/ui/SeekBarPreference.java:
```diff
@@ -44,6 +47,8 @@ public SeekBarPreference(Context context, AttributeSet attrs) {
         mMax = attrs.getAttributeIntValue(androidns, "max", 100);
         mMin = attrs.getAttributeIntValue(AnkiDroidApp.XML_CUSTOM_NAMESPACE, "min", 0);
         mInterval = attrs.getAttributeIntValue(AnkiDroidApp.XML_CUSTOM_NAMESPACE, "interval", 1);
+        mXLabel = attrs.getAttributeValue(AnkiDroidApp.XML_CUSTOM_NAMESPACE, "xlabel");
+        mYLabel = attrs.getAttributeValue(AnkiDroidApp.XML_CUSTOM_NAMESPACE, "ylabel");
```
Note: This would be better implemented as stated: https://stackoverflow.com/questions/46689210/android-how-to-use-string-resource-from-xml-file-in-property-of-customized-pref

```suggestion
        mXLabel = attrs.getAttributeResourceValue(AnkiDroidApp.XML_CUSTOM_NAMESPACE, "xlabel", 0);
        mYLabel = attrs.getAttributeResourceValue(AnkiDroidApp.XML_CUSTOM_NAMESPACE, "ylabel", 0);
```

david-allison-1 commented on AnkiDroid/src/main/java/com/ichi2/ui/SeekBarPreference.java:
```diff
@@ -66,6 +71,18 @@ protected View onCreateDialogView() {
                 LinearLayout.LayoutParams.WRAP_CONTENT));
 
+
+        if (mXLabel != null && mYLabel != null) {
```
```suggestion
        if (mXLabel != 0 && mYLabel != 0) {
```

david-allison-1 reviewed (approved):
Lovely stuff, thanks!

mikehardy reviewed (approved):
LGTM - I resolved the conflict related to #8586 which results in a little bit of a hash of a commit history on the exact PR here but it was one clean commit to begin with so squashing it back to one clean commit seems identical - ready to merge...

mikehardy commented:
This is the PR that just does not want to merge. While waiting for CI to go green on the last conflict resolution I did, it took another conflict from #8651 

Resolved, CI chewing again

mikehardy commented:
Got it merged before anything else could perturb it :-) :rocket:

Arnold2381 commented:
Accomplishment 😅😅
Thanks @mikehardy !!
